### ISSUE TITLE
Enhanced library

### ISSUE THREAD

**Issue by bragefuglseth** [sonnyp/Workbench#510]
It was mentioned in #505 that the library could use a rework. Here's an initial concept to get the ball rolling.

![library](https://github.com/sonnyp/Workbench/assets/91388039/e0714906-c949-4d75-803f-5157d1e23e11)

**Comment by bragefuglseth** [comment]
The icon library has icons for a lot of cases, but Workbench has a lot of demos to cover as well, so you have a point. I see three alternatives:
1. Drop the icons entirely. This would make the library less visually interesting, but it's the most scalable solution.
2. Only use icons for some demos, and find a nice way to display demos without them as well. I'm not sure how easy this would be design-wise.
3. De-emphasize the icon, choose one for each category and use that for apps. This might get repetitiv, but I can see it working nicely.

**Comment by bragefuglseth** [comment]
It seems like we need to flesh out the Demo Scene and tutorial concepts more to progress on this…

**Comment by vixalien** [comment]
> Maybe we could simply use a view switcher in Workbench for "main" categories.

Would that scale with many categories? Or are there a limited number of categories.

**Comment by BharatAtbrat** [comment]
The Gtk Flow box has horizontal tiling, is there an Adw counterpart for it? Gtk Grid also allows something similar

**Comment by bragefuglseth** [comment]
> is there an Adw counterpart for it?

What's stopping us from just using GtkFlowBox? There's no need for an equivalent in libadwaita :)

**Comment by vixalien** [comment]
I wouldn't recommend GtkFlowBox. GtkGrid or GtkGridView will be probably better.

**Comment by BharatAtbrat** [comment]
> I wouldn't recommend GtkFlowBox. GtkGrid or GtkGridView will be probably better.

Any specifics as to why?

**Comment by vixalien** [comment]
Yeah sorry, GtkFlowBox is just a glorified GridView that reflows automatically. This automatic reflow can get complicated as imo it's hard to predict.

Using GtkGridView allows you to control the reflow manually by adjusting the number of rows or columns.
